### PR TITLE
Add .editorconfig file

### PR DESCRIPTION
**Pull request by mfpierre**
### What does this PR do?

Add an [editorconfig](http://editorconfig.org) file

### Motivation

EditorConfig helps developers define and maintain consistent coding styles between different editors and IDEs.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# http://editorconfig.org
+
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
```
Comment (masci) [inline .editorconfig]: Maybe we should leave this unspecified to keep Windows folks happy?

Comment (mfpierre) [reply .editorconfig]: I guess they're used to this problem, I checked a bunch of project settings they all af this setting: https://github.com/editorconfig/editorconfig/wiki/Projects-Using-EditorConfig

Comment (masci) [reply .editorconfig]: More seriously, cross-platform projects seems to omit that settings, see https://github.com/Microsoft/vscode/blob/master/.editorconfig (also interesting for how they group targets with common settings)

**Review by masci — APPROVED** [review]
🍰

**Pull request merged**